hil/config.py

```diff
@@ -8,12 +8,50 @@
 import ConfigParser
 import logging.handlers
 import importlib
+from schema import Schema, Optional
 import os
 import sys
 
 cfg = ConfigParser.RawConfigParser()
 cfg.optionxform = str
 
+core_schema = {
+    Optional('general'): {
+        'log_level': str,
+        Optional('log_dir'): str,
+    },
+    Optional('auth'): {
+        Optional('require_authentication'):
+            lambda s: string_is_bool(s),
+    },
+    'headnode': {
+        'trunk_nic': str,
+        'base_imgs': str,
+        'libvirt_endpoint': str,
+    },
+    'client': {
+        Optional('endpoint'): str,
+    },
+    'database': {
+        'uri': str,
+    },
+    Optional('devel'): {
+        Optional('dry_run'):
+            lambda s: string_is_bool(s),
```

Comment:
The lambda is superfluous here (and elsewhere); you can just do `Optional('dry_run'): string_is_bool`. You'll have to move the definition of that function up though, so it's defined before core_schema.